### PR TITLE
Added changes for Pattern-1

### PR DESCRIPTION
**Pull request by BLasan**
- Removed MySQL PVC
- Added Headless Service for Worker-Analytics
- Added 'AllowAnalytics' field in types.go

### CODE REVIEW

File: artifacts/api-manager-artifacts/pattern-1/wso2am-p1-apim-1-conf.yaml [hunk 1]
```diff
@@ -131,7 +131,7 @@ data:
     #expiry_time = "2m"
 
     [apim.analytics]
-    enable = true
+    enable = "$env{ALLOW_ANALYTICS}"
```
Comment (pubudu538) [inline artifacts/api-manager-artifacts/pattern-1/wso2am-p1-apim-1-conf.yaml]: THis should be enable analytics.

**Review by pubudu538 — CHANGES_REQUESTED** [review]

**Comment by BLasan** [comment]
@pubudu538  Please review

File: pkg/apis/apim/v1alpha1/types.go [hunk 2]
```diff
@@ -122,12 +122,13 @@ type Artifacts struct {
...
 	// Here set UseMysql as string type to set true as default, and since using apiextensions.k8s.io/v1beta1
 	// and not using apiextensions.k8s.io/v1
-	UseMysql string    `json:"useMysql"`
-	Pattern  string    `json:"pattern"`
-	Replicas *int32    `json:"replicas"`
-	Profiles []Profile `json:"profiles"`
-	Service  Service   `json:"service"`
-	Expose   string    `json:"expose"`
+	UseMysql      string    `json:"useMysql"`
+	Allownalytics string    `json:"enableAnalytics"`
```
Comment (pubudu538) [inline pkg/apis/apim/v1alpha1/types.go]: Let's use the same enableAnalytics in here as well.

**Review by pubudu538 — APPROVED** [review]

File: pkg/controller/controller.go [hunk 3]
```diff
@@ -482,46 +483,62 @@ func (c *Controller) syncHandler(key string) error {
...
-		// Get analytics worker service name using hardcoded value
-		workerservice, err := c.servicesLister.Services(apimanager.Namespace).Get(workerServiceName)
-		// If the resource doesn't exist, we'll create it
-		if errors.IsNotFound(err) {
-			workerservice, err = c.kubeclientset.CoreV1().Services(apimanager.Namespace).Create(pattern1.WorkerService(apimanager))
+			// If the worker resource doesn't exist, we'll create it
+			if errors.IsNotFound(err1) {
+				y := pattern1.AssignApimAnalyticsWorkerConfigMapValues(apimanager, configmap, worknum)
+
+				// workerdeployment, err = c.kubeclientset.AppsV1().Deployments(apimanager.Namespace).Create(pattern1.WorkerDeployment(apimanager, y, worknum))
```
Comment (GDLMadushanka) [inline pkg/controller/controller.go:536]: remove comments

Comment (BLasan) [reply pkg/controller/controller.go:536]: fixed via https://github.com/BLasan/k8s-wso2am-operator/commit/8b163758448b29fec66bc8139ad184f405bf1f28

**Pull request merged**